---
title: "Vignette"
authors:   
    - "s252641 - Bhavya Gudhka"   
    - "s215065 - Henrik Servais Janssen Knudsen"   
    - "s225191 - Dhruva Gajanan Sakhare"   
    - "s253705 - Lun Suša"   
    - "s252623 - Felipe Alberto Morales Grimaldo" 
output: rmarkdown::pdf_document
vignette: >
  %\VignetteIndexEntry{Vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## GITHUB LINK

<https://github.com/rforbiodatascience25/group_26_package>

## OVERVIEW

The group26molbio package provides a collection of functions for performing basic operations in molecular biology — from DNA/RNA manipulation to sequence analysis. This vignette introduces each function and demonstrates how to use them effectively.

## INSTALLATION

```{r, message=FALSE}
devtools::install_github("rforbiodatascience25/group_26_package")
```

```{r setup}
library(group26molbio)
```

```{r, message=FALSE}
devtools::load_all()
```

## Functions

### randomDna()

**Purpose:**\
Generates a random DNA sequence of a specified length.\
This is useful for testing, simulations, or teaching exercises where you need arbitrary DNA sequences.

**Usage:**

```{r}
inputSize <- 12
dnaString <- randomDna(inputSize)
dnaString
```

### dnaToRna()

**Purpose:**\
Converts a DNA sequence into its corresponding RNA sequence by replacing all thymine (`T`) nucleotides with uracil (`U`).\
This is useful for simulating transcription or preparing sequences for RNA analysis.

**Usage:**

```{r}
rnaString <- dnaToRna(dnaString)
rnaString
```

### sequenceToCodons()

**Purpose:**\
Splits a RNA sequence into codons (3-base groups) starting from a given position defaulting at position 1.

```{r}
codons <- sequenceToCodons(rnaString)
```

### proteinTraduction()

**Purpose:**\
Translates a vector of codons into the corresponding protein sequence (amino acids).

**Usage:**

```{r}
aminoAcidString <- proteinTraduction(codons)
aminoAcidString
```

### aminoAcidsPlot()

**Purpose:**\
Generates a bar plot showing the frequency of each unique amino acid in a protein sequence.

**Usage:**

```{r}
aminoAcidsPlot(aminoAcidString)
```


## Future development

This package can be further expanded to include Motif Detection i.e. finding recurring amino acid patterns or conserved regions in sequences to find potential functional elements. The ploting can be extended to include multiple sequences to generate logo plots.

